src/Microsoft.Security.Utilities.Core/PreciselyClassifiedSecurityKeys/SEC101_200_CommonAnnotatedSecurityKey.cs

```diff
@@ -19,28 +19,37 @@ public CommonAnnotatedSecurityKey()
 
         public override IEnumerable<string> GenerateTruePositiveExamples()
         {
-            int count = 0;
             int attempts = 0;
 
-            while(true)
+            foreach (bool longForm in new[] { true, false })
             {
-                foreach (bool longForm in new[] { true, false })
+                while (true) 
                 {
                     char testChar = (char)('a' + attempts++);
-                    string example = IdentifiableSecrets.GenerateCommonAnnotatedTestKey(IdentifiableSecrets.VersionTwoChecksumSeed,
-                                                                                        "TEST",
-                                                                                        customerManagedKey: true,
-                                                                                        platformReserved: null,
-                                                                                        providerReserved: null,
-                                                                                        longForm: false,
-                                                                                        testChar);
 
-                    if (example == null) { continue; }
+                    if (testChar == '{')
+                    {
+                        break;
+                    }
+
+                    string example;
 
-                    if (++count == 20)
+                    try
+                    { 
+                        example = IdentifiableSecrets.GenerateCommonAnnotatedTestKey(IdentifiableSecrets.VersionTwoChecksumSeed,
+                                                                                     "TEST",
+                                                                                     customerManagedKey: true,
+                                                                                     platformReserved: null,
+                                                                                     providerReserved: null,
+                                                                                     longForm,
+                                                                                     testChar);
+                    }
+                    catch (InvalidOperationException)
```

Comment:
>InvalidOperationException [](http://example.com/codeflow?start=27&length=25)

This seems wrong. The real problem here is that someone specified a test char that does not comprise a base62 character. We should change the API to perform this check and raise `ArgumentException`. If you prefer not to be slowed down by this suggestion, feel free to check in now and defer that change.

Reply:
I agree this is a better approach, since it will avoid a bunch of processing on an illegal character before the argument is thrown. I'll track this and make follow-on change.

Reply:
I left this open---it deserves a follow on PR as it's an API change